spec/controllers/projects_controller_spec.rb

```diff
@@ -144,6 +144,20 @@
     end
   end
 
+  describe "#toggle_archive" do
```

Comment:
Question: Would it make sense to have this type of test as a system test instead of a controller test?

Reply:
It's a good suggestion. @fbuys 
The only thing I would do is, keep the controller test as well and add the relevant feature test case for it.

While adding, I realized that the feature test for "archive a project" already exists, so I have just added the additional checks in that and added a new test case for "unarchive a project".